bundles/io/org.eclipse.smarthome.io.net/src/main/java/org/eclipse/smarthome/io/net/http/HttpUtil.java

```diff
@@ -388,17 +419,18 @@ public static RawType downloadImage(String url, boolean scanTypeInContent, long
      * @param scanTypeInContent true to allow the scan of data to determine the content type if not found in the headers
      * @param maxContentLength the maximum data size in bytes to trigger the download; any negative value to ignore the
      *            data size
+     * @param timeout the socket timeout in milliseconds to wait for data
      * @return a RawType object containing the downloaded data, null if the content type does not match the expected
      *         type or the data size is too big
      */
     public static RawType downloadData(String url, String contentTypeRegex, boolean scanTypeInContent,
-            long maxContentLength) {
+            long maxContentLength, int timeout) {
```

Comment:
Maybe I miss something here, but isn't this breaking the API? Keep in mind - the purpose of this class is (as I learned) primarily to be exposed in scripts and rules. Therefore this would breaking endusers' code here.

Reply:
You are right this method does not seem have a backwards compatible variant, missed that :-)

Reply:
I will fix that.